### PR TITLE
Fix 78249

### PR DESCRIPTION
**Pull request by lola27**

### CODE REVIEW

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    volblocksize_placeholder: T("Block Size"),
+    volblocksize_tooltip: T(""),
+
+    usefor_placehodler: T("What are you using this for"),
+    usefor_tooltip: T(""),
+
+    step2_label: T("Portal"),
+
+    portal_placeholder: T("Portal"),
+    portal_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Select an existing portal or choose <i>Create New</i> to configure a new portal.`

(new text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    discovery_authmethod_placeholder: T("Discovery Auth Method"),
+    discovery_authmethod_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `iSCSI(https://ixsystems.com/documentation/freenas/11.2/sharing.html?runningversion=FreeNAS-11.2-U2-INTERNAL96#block-iscsi) supports multiple authentication methods that are used by the target to discover valid devices. None allows anonymous discovery while CHAP and Mutual CHAP require authentication.`

(From Portal > Discovery Auth Method help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    discovery_authgroup_placeholder: T("Discovery Auth Group"),
+    discovery_authgroup_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Select a user created in Authorized Access if the Discovery Auth Method is set to CHAP or Mutual CHAP.`

(From Portal > Discover Auth Group help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    ip_placeholder: T("IP"),
+    ip_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Select the IP address associated with an interface or the wildcard address of 0.0.0.0 (any interface).`

(From Portal > IP Address help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    port_placeholder: T("Port"),
+    port_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `TCP port used to access the iSCSI target. Default is 3260.`

(From Portal > Port help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    auth_placeholder: T('Authorized Access'),
+    auth_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Select an existing user configuration or <i>Create New</i> to view options for a new user configuration.`

(New text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+    tag_placeholder: T("Group ID"),
+    tag_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Allows different groups to be configured with different authentication profiles. Example: all users with a group ID of 1 will inherit the authentication profile associated with Group 1.`

(From Authorized Access > Group ID help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    user_placeholder: T("User"),
+    user_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Enter name of user account to use for CHAP authentication with the user on the remote system. Many initiators default to the initiator name as the user.`

(From Authorized Access > User help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+    secret_placeholder: T("Secret"),
+    secret_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Enter a password for the User. Must be between 12 and 16 characters.`

(From Authorized Access > Secret help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+    secret_confirm_placeholder: T("Secret (Confirm)"),
+    secret_confirm_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Retype the <i>Secret</i>.`

(New text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+    step3_label: T("Initiator"),
+
+    initiators_placeholder: T("Initiators"),
+    initiators_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Enter <i>ALL</i> or a list of initiator hostnames separated by spaces.`

(From Initiators > Initiators help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+    auth_network_placeholder: T("Authorized Networks"),
+    auth_network_tooltip: T(""),
```
Comment (Mrt134) [inline src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: `Network addresses that can use this initiator. Enter <i>ALL</i> or list network addresses with a CIDR mask. Separate multiple addresses with a space: <i>192.168.2.0/24 192.168.2.1/12</i>.`

(From Initiators > Authorized Networks help text)

File: src/app/helptext/sharing/iscsi/iscsi-wizard.ts [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+    name_tooltip: T(""),
+
+    type_placeholder: T("Type"),
+    type_tooltip: T(""),
+
+    file_placeholder: T("File"),
+    file_tooltip: T(""),
+
+    filesize_placeholder: T("Filesize"),
+    filesize_tooltip: T(""),
```
[the remark this answers is not in the document]
Comment (wblock) [reply src/app/helptext/sharing/iscsi/iscsi-wizard.ts]: "When the file already exists, enter a size of <i>0</i> to use the actual file size. For new files, enter the size of the file to create."

**Comment by lola27** [comment]
@pkgdemon the available disk choice issue is fixed by Middleware team https://github.com/freenas/freenas/pull/2740

File: src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts [hunk 14]
```diff
@@ -0,0 +1,866 @@
...
+                    name: 'volblocksize',
+                    placeholder: helptext.volblocksize_placeholder,
+                    tooltip: helptext.volblocksize_tooltip,
+                    isHidden: true,
+                },
+                // use for group
+                {
+                    type: 'select',
+                    name: 'usefor',
+                    placeholder: helptext.usefor_placehodler,
```
Comment (wblock) [inline src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts]: Typo: s/placehodler/placeholder/

**Pull request merged**